client/src/services/api.js

```diff
@@ -1,3 +1,11 @@
-const fetchData = () => fetch('/api/v1').then(res => res.json());
-
-export default { fetchData };
+export default {
+  login: data => {
+    return fetch('/api/v1/login', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    }).then(res => res.json());
```

Comment:
```suggestion
    }).then(res => {
        if (!res.ok) {
            throw Error(res.message ? res.message : 'Login failed');
        }
        return res.json();
    });
```
the `if (!res.ok) {` checks the HTTP status code, evaluates to true if it's anything other than success (200). Every API in this module should utilize this approach (you can extract it into a function and use `.then(handleResp)`)